### DISCUSSION TITLE
Thảo luận cho báo cáo bài 4

### DISCUSSION THREAD

**Discussion by trthann** [trthann/L-p-10A06---Nh-m-02#5] [General]
Ai có cần giúp gì hong za ?

**Comment by trthann** [comment]
Mọi người vô teams xem tiêu chí chấm điểm của thầy nha

**Comment by YDinhC06** [comment]
Tí tối t làm rồi gửi lên nha An

**Comment by YDinhC06** [comment]
Mấy bạn ơi ai giúp tui câu 3 zới, tui chưa tìm được

**Comment by YDinhC06** [comment]
•CÂU 4:
Có 2 cách biểu diễn thuật toán:
- Sử dụng cách liệt kê: để nêu ra tuần tự các thao tác cần tiến hành.
- Sử dụng sơ đồ khối: để mô tả thuật toán. 
-Ví dụ câu 3 thuật toán được diễn tả theo cách liệt kê
(Thấy thiếu gì mấy bạn cứ nói nha)

**Comment by PhuongNgannLe** [comment]
Câu 2:
Để từ Input của bài toán, máy tính tìm cho ta Output, chúng ta cần chứng minh sự tồn tại của lời giải vaf không cần chỉ ra một cách tường minh cách tìm lời giải đó. Việc chỉ chỉ ra một cách tìm ra Output của bài toán được gọi là một thuật toán (alogorithm).
Nhận xét: Thuật toán là các bước tìm ra Output dựa vào Input với sự thực hiện của máy tính.
Khái niệm: Thuật toán là một dãy hữu hạn các thao tác được sắp xếp theo một trình tự xác định sao cho sau khi thực hiện dãy thao tác ấy, từ Input của bài toán, ta nhận được Output cần tìm.
Các tính chất của thuật toán:
- Tính dừng: Thuật toán phải kết thúc sau một số hữu hạn lần thực hiện các thao tác
- Tính xác định: Sau khi thực hiện một thao tác thì hoặc là thuật toán kết thúc hoặc là có đúng một thao tác xác định để được thực hiện tiếp theo
- Tính đúng đắn: Sau khi thuật toán kết thúc, ta phải nhận được Output cần tìm
Ví dụ: Với thuật toán tìm Max đã xét:
- Tính dừng: Vì giá trị của i mỗi lần tăng lên 1 nên sau N lần thì i > N, khi đó kết quả phép so sánh ở bước 3 xác định việc đưa ra giá trị Max rồi kết thúc.
- Tính xác định: Thứ tự thực hiện các bước của thuật toán được mặc định là tuần tự nên sau bước 1 là bước 2, sau bước 2 là bước 3. Kết quả các phép so sánh trong bước 3 và bước 4 đều xác định duy nhất bước tiếp theo cần thực hiện.
- Tính đúng đắn: Vì thuật toán so sánh Max với từng số hạng của dãy số và thực hiện Max <— a^i nếu a^i > Max nên sau khi so sánh hết N số hạng của dãy thì Max là giá trị lớn nhất.

**Comment by PhuongNgannLe** [comment]
Câu 5:
Bài toán giải phương trình bậc hai: ax² + bx + c = 0 (a khác 0)
Input: Các số thực a, b, c (a khác 0)
Output: Tất cả các số thực x thoả mãn
ax² + bx + c = 0
Ở đây, Output có thể là một hoặc hai số thực hoặc câu trả lời không có số thực nào như vậy.

**Comment by trthann** [comment]
t thấy bài r nha

**Comment by duyanhthptphunhuan** [comment]
Câu 6
-Có 2 cách mô tả thuật toán trên câu 5 là:
+ Liệt kê bước	
+ Sơ đồ khối   
        -Mô tả thuật toán bằng cách liệt kê:
Bước 1: Nhập 3 số a,b,c
Bước 2: d	(b*b-4*b*c)
Bước 3.
        +Nếu d < 0 thì đưa ra thông báo phương trình vô nghiệm rồi kết thúc.
        +Nếu d=0 thì đưa ra thong báo phương trình có một nghiệm và một tính nghiệm
x=-b/(2*a), rồi kết thúc
 + Nếu d>0 thì đưa ra thong báo phương trình có hai nghiệm phân biệt, tính nghiệm x1=(-b+√d)/(2*a) và x2=(- b-√d)/(2*a), rồi kết thúc

**Comment by YDinhC06** [comment]
Câu 3
Thuật toán 2 được sử dụng để giải thuật toán. 
Vì do đề yêu cầu vẽ tam giác vuông mà thuật toán 1 chỉ vẽ tam giác,nếu là vẽ tam giác gì thì cũng có thể kết thúc thuật toán chưa đủ yêu cầu của đề là tam giác vuông.
